diff --git a/node_modules/@cap-js/cds-test/lib/naxios.js b/node_modules/@cap-js/cds-test/lib/naxios.js
index f08dc96..7f358fe 100644
--- a/node_modules/@cap-js/cds-test/lib/naxios.js
+++ b/node_modules/@cap-js/cds-test/lib/naxios.js
@@ -55,8 +55,10 @@ class Naxios {
     if (data) o.body =
       typeof data === 'string' ? data :
       data instanceof Readable ? data :
+      Buffer.isBuffer(data) ? data :
       JSON.stringify(data)
-    if (typeof data === 'object' && !o.headers.has('Content-Type')) o.headers.set('Content-Type', 'application/json')
+    if (typeof data === 'object' && !Buffer.isBuffer(data) && !(data instanceof Readable) && !o.headers.has('Content-Type'))
+      o.headers.set('Content-Type', 'application/json')
     if (!url.startsWith('http')) url = (o.baseURL||'') + (o.path||'') + (url[0]==='/'?'':'/') + url
     if (params) url += '?' + new URLSearchParams (params)
     o.url = url
@@ -77,7 +79,7 @@ class Naxios {
       case 'arraybuffer': return res.arrayBuffer().then(Buffer.from)
     }
     let ct = res.headers.get('content-type')
-    if (/stream|image|pdf|tar/.test(ct)) return res.body
+    if (/stream|image|pdf|tar|octet-stream/.test(ct)) return res.arrayBuffer().then(Buffer.from)
     if (/xml/.test(ct)) return res.text()
     else return res.text().then(x => {
       try { return JSON.parse(x) }
